src/main/cpp/tests/timezones.cpp

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "timezones.hpp"
+
+#include <climits>
+
+#include <cudf/wrappers/timestamps.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/iterator_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
+
+using seconds_col =
+    cudf::test::fixed_width_column_wrapper<cudf::timestamp_s, cudf::timestamp_s::rep>;
+
+using millis_col =
+    cudf::test::fixed_width_column_wrapper<cudf::timestamp_ms, cudf::timestamp_s::rep>;
+
+using micros_col =
+    cudf::test::fixed_width_column_wrapper<cudf::timestamp_us, cudf::timestamp_s::rep>;
+
+class TimeZoneTest : public cudf::test::BaseFixture {
+protected:
+  void SetUp() override
+  {
+    transitions = make_transitions_table();
+  }
+  std::unique_ptr<cudf::table> transitions;
+
+private:
+  std::unique_ptr<cudf::table> make_transitions_table()
+  {
+    auto instants_from_utc_col = cudf::test::fixed_width_column_wrapper<int64_t>(
```

Comment:
Since we are using type aliasing like `seconds_col`, please do similarly for `int64_t` such as `int64s_col`.